### PR TITLE
Update to RC1

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

**Comment by dagood** [comment]
The error aspnetcore is hitting now is this, during restore:

```
/work/artifacts/src/aspnetcore.ecdcc752d4639061c2c49727ee77a1039bbbca22/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj error MSB4236:
  The SDK 'Microsoft.NET.Sdk.BlazorWebAssembly' specified could not be found [/work/artifacts/src/aspnetcore.ecdcc752d4639061c2c49727ee77a1039bbbca22/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj]
```

which starts in evaluation:

```
Project Name=/work/artifacts/src/aspnetcore.ecdcc752d4639061c2c49727ee77a1039bbbca22/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj id:583 File=/work/artifacts/src/aspnetcore.ecdcc752d4639061c2c49727ee77a1039bbbca22/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj
Resolving SDK 'Microsoft.NET.Sdk.BlazorWebAssembly'...
[SourceBuiltSdkResolver] Looking for SDK Microsoft.NET.Sdk.BlazorWebAssembly. Detected config(s) in env: 'ARCADE' for 'Microsoft.DotNet.Arcade.Sdk/5.0.0-beta.20431.1' at '/work/Tools/source-built/Microsoft.DotNet.Arcade.Sdk/'
Errors
  /work/artifacts/src/aspnetcore.ecdcc752d4639061c2c49727ee77a1039bbbca22/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj [SourceBuiltSdkResolver] 0 overrides found for 'Microsoft.NET.Sdk.BlazorWebAssembly'
```

The `SourceBuiltSdkResolver` "error" is a red herring. The "error" is just the source build resolver saying why it didn't do an override. It's expected that it won't do an override here.

If the .NET SDK goes on to find the SDK itself, there's no problem. But when the SDK fails to find `Microsoft.NET.Sdk.BlazorWebAssembly` in any other way (nuget restore, bundled in the SDK), it then prints out the messages from `SourceBuiltSdkResolver` as errors without telling us about the other ways it failed to find it.

I *think* this tracks better error messages from the built-in resolver: https://github.com/dotnet/sdk/issues/9655.

---

In this case, `Microsoft.NET.Sdk.BlazorWebAssembly` is present in .NET SDK `5.0.100-rc.1` but not `5.0.100-preview.8`, so we need to upgrade. I confirmed that in a standalone build of aspnetcore, it uses the copy that's distributed in the rc1 .NET SDK. (Haven't tried out changing source-build to rc1 yet.)

File: patches/arcade/0008-Use-SystemReflectionMetadata-from-package-flow.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+From 3066411067e8cbe2446344999d01cddad652210c Mon Sep 17 00:00:00 2001
+From: dseefeld <dseefeld@microsoft.com>
+Date: Tue, 22 Sep 2020 15:32:53 +0000
+Subject: [PATCH] Use SystemReflectionMetadata from package flow in offline
+
+---
```
Comment (dagood) [inline patches/arcade/0008-Use-SystemReflectionMetadata-from-package-flow.patch:6]: Why?

File: patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch [hunk 2]
```diff
@@ -0,0 +1,52 @@
+From 2cdd560b4029623d27a7133b2651a76c2b209794 Mon Sep 17 00:00:00 2001
+From: dseefeld <dseefeld@microsoft.com>
+Date: Thu, 8 Oct 2020 22:19:34 +0000
+Subject: [PATCH] Add tools folder to crossgen exclusions for
+ MS.NET.Build.Extensions
+
+---
```
Comment (dagood) [inline patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch:7]: Why? Should the upstream repo be changed to not produce a `tools` folder instead, is this a diff vs. the Microsoft-built package? (File issue for followup with more info if it exists?)

File: patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
++      <RemainingFiles Remove="$(SdkOutputDirectory)Microsoft\Microsoft.NET.Build.Extensions\tools\net*\**\*" />
+ 
+       <!-- Don't CrossGen bundled DotnetTools -->
+       <RemainingFiles Remove="$(SdkOutputDirectory)DotnetTools\**\*" />
+@@ -98,7 +99,7 @@
+       <!-- FSharp.Build.dll causes the FSharp folder to be included. Remove it, as we don't want other FSharp dlls being included in the crossgen. -->
+       <RemainingFolders Remove="$(PublishDir)FSharp\**\*" />
+ 
+-      <!-- RemainingFolders should be ordered so that dependencies are first looked up in the leaf folders. 
++      <!-- RemainingFolders should be ordered so that dependencies are first looked up in the leaf folders.
```
Comment (dagood) [inline patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch:46]: Undo unnecessary whitespace changes in this patch?

File: repos/aspnetcore.proj [hunk 4]
```diff
@@ -16,6 +16,10 @@
     <BuildCommandArgs>$(BuildCommandArgs) --arch $(Platform)</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:BuildNodeJs=false</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:SourceBuildRuntimeIdentifier=$(OverrideTargetRid)</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) /p:MicrosoftNetFrameworkReferenceAssembliesVersion=1.0.0</BuildCommandArgs>
+    <!-- aspnetcore requires some types in the RC1 version of this package.  This is temporary until RC1 versions
+        are included in previously source-built packages -->
+    <BuildCommandArgs>$(BuildCommandArgs) /p:MicrosoftNETCoreAppRefPackageVersion=5.0.0-rc.1.20451.14</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:UseAppHost=false</BuildCommandArgs>
```
Comment (dagood) [inline repos/aspnetcore.proj:23]: Add comment to the first one, put these two props after the rest of the args to stand them apart from the more typical build configuration changes?

Does the PVP not work for these?

**Review by dagood — APPROVED** [review]

File: patches/arcade/0008-Use-SystemReflectionMetadata-from-package-flow.patch (hunk 1 above)
Comment (dseefeld) [reply patches/arcade/0008-Use-SystemReflectionMetadata-from-package-flow.patch]: Removed the patch

File: patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch (hunk 2 above)
Comment (dseefeld) [reply patches/installer/0007-Add-tools-folder-to-crossgen-exclusions-for-MS.NET.B.patch]: Removed the patch.  There was an issue in how sdk builds MS.NET.Build.Extensions.  I made a change there.

File: repos/aspnetcore.proj (hunk 4 above)
Comment (dseefeld) [reply repos/aspnetcore.proj]: Done.  The PVP doesn't work because the PVP has the previously source-built version, which is a preview8 version.  When previously source-built is updated, this can be removed.

**Comment by dseefeld** [comment]
@crummel PTAL

**Review by crummel — APPROVED** [review]

**Comment by dseefeld** [comment]
OSX build issue is https://github.com/dotnet/source-build/issues/1819.
